### ISSUE TITLE
`build web views` should be a prelaunch task for running the extension

### ISSUE THREAD

**Issue by amunger** [microsoft/vscode-jupyter#11997]
when running the extension from source, we don't check if web views are built, so the error renderer and other web views won't load correctly until that build task is run.

**Comment by Tyriar** [comment]
You might be better off doing a compound task like this:

https://github.com/xtermjs/xterm.js/blob/a351f5758a5126308b90d60b604b528462f6f051/.vscode/tasks.json#L43-L44

**Comment by roblourens** [comment]
I actually kind of disagree with that- 99% of the time, I don't need it, and it's another slow task to run. But I can work around it myself if I'm outvoted.

**Comment by amunger** [comment]
maybe there's a more helpful notification we can give to developers then? It's currently pretty hard to know what went wrong or how to fix it without being able to just ask the team.

**Comment by roblourens** [comment]
I guess as an OSS project we should aim for the best defaults for non-experts

**Comment by rebornix** [comment]
Maybe if we move to esbuild we can have a compound task

**Comment by DonJayamanne** [comment]
> actually kind of disagree with that- 99% of the time, I don't need it, and it's another slow task to run. But I can work around it myself if I'm

i agree 110% 😉

however, i have a pr in progress that would make this irrelevant. i am using esbuild, and that completes very quickly, fast enough to make it a pre launch task.
will submit that pr in debt week, didn't have time to polish it before the end of last debt week

**Comment by DonJayamanne** [comment]
I'll assign this to myself as i hand that work almost ready to go

**Comment by Tyriar** [comment]
I don't think the project has a default build task currently, so this would just be making it the default one that's also recommended in the docs. You are still free to run whatever individual task you want.

Also does it add that much overhead? Wouldn't it just be in a separate task terminal that's run in parallel?

**Comment by DonJayamanne** [comment]
> Also does it add that much overhead? 

Currently yes (its slow), that's why I don't like it

> Wouldn't it just be in a separate task terminal that's run in parallel?

yes we could, one of the reasons I've not been a fan is this slows down debugging (as we need to wait for this task to complete).

**Comment by DonJayamanne** [comment]
Done
